cmd/contour/servecontext_test.go

```diff
@@ -191,9 +200,167 @@ leaderelection:
 	}
 }
 
+// testdata for this test case can be re-generated by running:
+// make gencerts
+// cp certs/*.pem cmd/contour/testdata/X/
+func TestServeContextCertificateHandling(t *testing.T) {
+	tests := map[string]struct {
+		serverCredentialsDir string
+		clientCredentialsDir string
+		expectedServerCert   string
+		expectError          bool
+	}{
+		"successful TLS connection established": {
+			serverCredentialsDir: "testdata/1",
+			clientCredentialsDir: "testdata/1",
+			expectedServerCert:   "testdata/1/contourcert.pem",
+			expectError:          false,
+		},
+		"rotating server credentials returns new server cert": {
+			serverCredentialsDir: "testdata/2",
+			clientCredentialsDir: "testdata/2",
+			expectedServerCert:   "testdata/2/contourcert.pem",
+			expectError:          false,
+		},
+		"fail to connect with bad client certificate": {
```

Comment:
IIUC, this fails because the client and server don't agree on the CA root certificate to expect? Let's update the description.

Reply:
Yes exactly, I updated the description.